Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -272,15 +290,19 @@ static OSStatus outputRenderCallBack(void *inRefCon,
 {
     OCTAudioEngine *myEngine = (__bridge OCTAudioEngine *)inRefCon;
 
-    int bytesToCopy = ioData->mBuffers[0].mDataByteSize;
+    UInt32 bytesToCopy = ioData->mBuffers[0].mDataByteSize;
     SInt16 *targetBuffer = (SInt16 *)ioData->mBuffers[0].mData;
 
     int32_t availableBytes;
-    SInt16 *buffer = TPCircularBufferTail(&myEngine->_buffer, &availableBytes);
-    int32_t sampleCount = MIN(bytesToCopy, availableBytes);
+    SInt16 *buffer = TPCircularBufferTail(&myEngine->_incomingBuffer, &availableBytes);
+    UInt32 sampleCount = MIN(bytesToCopy, availableBytes);
 
+    if (bytesToCopy > availableBytes) {
+        memset(targetBuffer, 0, bytesToCopy);
```

Comment:
Yea that was the way I originally wanted to do it, but I wasn't sure how to copy and turn the rest to zeroes? Or are they default zeroes?

Right now how it works is that if there isn't it enough buffers to copy over, it will skip the cycle and consume it on the next run when it has accumulated enough. Anyways I'm not sure if it makes any difference? What do you think?